src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+      get<::Tags::Tempa<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+  auto& two_index_constraint =
+      get<::Tags::Tempia<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+
+  *inverse_spacetime_metric = determinant_and_inverse(spacetime_metric).second;
+  gr::spatial_metric(make_not_null(&spatial_metric), spacetime_metric);
+  *inverse_spatial_metric = determinant_and_inverse(spatial_metric).second;
+  raise_or_lower_index(unit_interface_normal_vector, normal_covector,
+                       *inverse_spatial_metric);
+
+  gr::shift(make_not_null(&shift), spacetime_metric, *inverse_spatial_metric);
+  //   gr::lapse(make_not_null(&lapse), shift, spacetime_metric);
+  gr::spacetime_normal_vector(spacetime_unit_normal_vector, lapse, shift);
+  gr::spacetime_normal_one_form(make_not_null(&spacetime_unit_normal_one_form),
+                                lapse);
+
+  GeneralizedHarmonic::extrinsic_curvature(
+      extrinsic_curvature, *spacetime_unit_normal_vector, pi, phi);
+
+  GeneralizedHarmonic::three_index_constraint(three_index_constraint,
+                                              d_spacetime_metric, phi);
+  if (LIKELY(Dim == 3)) {
+    GeneralizedHarmonic::four_index_constraint(four_index_constraint, d_phi);
+  } else if (UNLIKELY(Dim == 2)) {
+    for (size_t a = 0; a <= Dim; ++a) {
+      for (size_t b = 0; b <= Dim; ++b) {
+        four_index_constraint->get(0, a, b) =
+            d_phi.get(0, 1, a, b) - d_phi.get(1, 0, a, b);
+        four_index_constraint->get(1, a, b) =
+            -four_index_constraint->get(0, a, b);
+      }
+    }
+  } else {
+    std::fill(four_index_constraint->begin(), four_index_constraint->end(), 0.);
+  }
+
+  gr::interface_null_normal(incoming_null_one_form,
+                            spacetime_unit_normal_one_form, normal_covector,
+                            -1.);
+  gr::interface_null_normal(outgoing_null_one_form,
+                            spacetime_unit_normal_one_form, normal_covector,
+                            1.);
+  gr::interface_null_normal(incoming_null_vector, *spacetime_unit_normal_vector,
+                            *unit_interface_normal_vector, -1.);
+  gr::interface_null_normal(outgoing_null_vector, *spacetime_unit_normal_vector,
+                            *unit_interface_normal_vector, 1.);
+
+  gr::transverse_projection_operator(projection_ab, spacetime_metric,
+                                     spacetime_unit_normal_one_form,
+                                     normal_covector);
+  gr::transverse_projection_operator(
+      projection_Ab, *spacetime_unit_normal_vector,
+      spacetime_unit_normal_one_form, *unit_interface_normal_vector,
+      normal_covector);
+  gr::transverse_projection_operator(projection_AB, *inverse_spacetime_metric,
+                                     *spacetime_unit_normal_vector,
+                                     *unit_interface_normal_vector);
+
+  const auto dt_char_fields = characteristic_fields(
```

Comment:
same question about extra time derivatives for tensor factors in the characteristic computaiton

Reply:
See [above](https://github.com/sxs-collaboration/spectre/pull/3172#discussion_r646623671)